### DISCUSSION TITLE
R generated table cross reference on GFM

### DISCUSSION THREAD

**Discussion by Yoshinobu-Ishizaki** [quarto-dev/quarto-cli#895] [Q&A]
When cross referencing table generated by R, "#| label" seems not working on gfm format.
````
---
title: "Untitled"
format: gfm
---

```{r}
#| label: tbl-table1
#| tbl-cap: Cars

library(knitr)
kable(head(cars))
```
````

Just generates table w/o caption and reference. 

```
| speed | dist |
|------:|-----:|
|     4 |    2 |
|     4 |   10 |
|     7 |    4 |
|     7 |   22 |
|     8 |   16 |
|     9 |   10 |
```

Code below works fine, but perhaps this is not consistent with Quarto design?

````
---
title: "Untitled"
format: gfm
---

```{r}
library(knitr)
kable(head(cars), caption = "Cars {#tbl-table1}")
```

Ref @tbl-table1
````

**Comment by jjallaire** [comment]
Thanks for reporting this. I have resolved this here: https://github.com/quarto-dev/quarto-cli/commit/13ee4741a7280a4cd4d5dcc6283caf4451612f89

Note that crossrefs to tables aren't a perfect fit for `gfm` as it requires that a `<div>` wrap the table to provide an id target. Here is the generated markdown:

````
<div id="tbl-table1">

| speed | dist |
|------:|-----:|
|     4 |    2 |
|     4 |   10 |
|     7 |    4 |
|     7 |   22 |
|     8 |   16 |
|     9 |   10 |

Table 1: Cars

</div>

Ref [Table 1](#tbl-table1)
````

This may or may not actually work in practice (e.g. if the markdown renderer strips divs or adds an id prefix to divs as GH does).

**Reply by Yoshinobu-Ishizaki** [reply]
Thank you ! Somehow it partially works on Gitlab.
